src/nodes.ts

```diff
@@ -0,0 +1,53 @@
+import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+
+export interface Nodes extends AsyncIterable<JsonLdObj> {
+    count(): Promise<number>;
+
+    get(id: Iri): Promise<JsonLdObj>;
+
+    set(article: JsonLdObj): Promise<void>;
+
+    delete(id: Iri): Promise<void>;
+
+    has(id: Iri): Promise<boolean>;
+}
+
+export default class InMemoryNodes implements Nodes {
+    private nodes: { [key: string]: JsonLdObj } = {};
+
+    private i = 0;
+
+    [Symbol.asyncIterator](): AsyncIterator<JsonLdObj> {
+      const isDone = Object.keys(this.nodes)[this.i] === undefined;
```

Comment:
This now works when no nodes are set but we still experience the fatal error when we set any nodes.